artemis-server/src/test/java/org/apache/activemq/artemis/core/server/cluster/impl/ClusterConnectionImplMockTest.java

```diff
@@ -85,6 +93,53 @@ public void testRemvalOfLocalParameters() throws Exception {
 
    }
 
+   @Test
+   public void testNullPrimaryOnNodeUp() throws Exception {
+      TransportConfiguration tc = new TransportConfiguration();
+      tc.setFactoryClassName("mock");
+      tc.getParams().put(TransportConstants.LOCAL_ADDRESS_PROP_NAME, "localAddress");
+      tc.getParams().put(TransportConstants.LOCAL_PORT_PROP_NAME, "localPort");
+
+      ArtemisExecutor executor = ArtemisExecutor.delegate(Executors.newSingleThreadExecutor(ActiveMQThreadFactory.defaultThreadFactory(getClass().getName())));
+
+      ClusterConnectionImpl cci = new ClusterConnectionImpl(
+         null, //final ClusterManager manager,
+         new TransportConfiguration[]{tc}, //final TransportConfiguration[] staticTranspConfigs,
+         null, //final TransportConfiguration connector,
+         null, //final SimpleString name,
+         null, //final SimpleString address,
+         0, //final int minLargeMessageSize,
+         0L, //final long clientFailureCheckPeriod,
+         0L, //final long connectionTTL,
+         0L, //final long retryInterval,
+         0, //final double retryIntervalMultiplier,
+         0L, //final long maxRetryInterval,
+         0, //final int initialConnectAttempts,
+         0, //final int reconnectAttempts,
+         0L, //final long callTimeout,
+         0L, //final long callFailoverTimeout,
+         false, //final boolean useDuplicateDetection,
+         null, //final MessageLoadBalancingType messageLoadBalancingType,
+         0, //final int confirmationWindowSize,
+         0, //final int producerWindowSize,
+         () -> executor,//final ExecutorFactory executorFactory,
+         new MockServer(), //final ActiveMQServer server,
+         null, //final PostOffice postOffice,
+         null, //final ManagementService managementService,
+         null, //final ScheduledExecutorService scheduledExecutor,
+         0, //final int maxHops,
+         new FakeNodeManager(UUIDGenerator.getInstance().generateStringUUID()), //final NodeManager nodeManager,
+         null, //final String clusterUser,
+         null, //final String clusterPassword,
+         true, //final boolean allowDirectConnectionsOnly,
+         0, //final long clusterNotificationInterval,
+         0 //final int clusterNotificationAttempts)
```

Comment:
Does the commented out parameter code really add enough to warrant keeping the cruft? It will just go stale as people rework the impl. Even assigning all the actual variables would probably look nicer (if more verbose..though many could probably be multi-assigned to reduce line count). The values are mostly essentially 0/null/false except a small subset. Maybe just comment ones that are particularly noteworthy?